src/mpb.cpp

```diff
@@ -640,6 +645,7 @@ void fields::get_eigenmode_coefficients(dft_flux flux,
 
       double vg=get_group_velocity(mode_data);
       if (vgrp) vgrp[nb*num_freqs + nf]=vg;
+      if (kpoints) kpoints[nb*num_freqs + nf]=get_k(mode_data);
```

Comment:
This is in the reciprocal basis, I think. You need to multiply by G (probably when mode_data is initialized?) to convert to Cartesian